### DISCUSSION TITLE
Could not able to understand subscription docs

### DISCUSSION THREAD

**Discussion by Abdullah-onGit** [the-guild-org/apollo-angular#1696] [Q&A]
Hi,

I couldn't able to follow this docs [https://apollo-angular.com/docs/data/subscriptions#client-setup](subscriptions in apollo).

```
import {WebSocketLink} from 'apollo-link-ws';

const wsLink = new WebSocketLink({
  uri: `ws://localhost:5000/graphql`,
  options: {
    reconnect: true,
    connectionParams: {
      authToken: user.authToken,
    },
  },
});
```
what is the above code snippet? no explanation ?
totally, I couldn't able to develop subscription part by following the documentation

can anyone recommend any step by step tutorial how to implement subscriptions with latest angular framework

**Comment by Urigo** [comment]
@Abdullah-onGit could you create a repo with your try?
then we could jump in, help you fix it and later you can change the docs to be correct for everyone else?
Thanks!
